### PR TITLE
DATAUP-314: test modal elimination

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Removing more DOM contamination, plus a wee bit of inline style removal to boot.

SonarCloud has failed on duplication as the tests for the taxonomyRef and select2Object inputs are extremely similar (the widgets themselves are very similar).

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-314
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to see that nothing has changed.

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/kbaseNarrative.js [hunk 1]
```diff
@@ -333,17 +333,17 @@ define([
      * after there's a visible DOM element for it to render in.
      */
     Narrative.prototype.initSharePanel = function () {
-        let sharePanel = $(
+        const sharePanel = $(
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/kbaseNarrative.js:336]: eslint fixes

File: kbase-extension/static/kbase/js/kbaseNarrativePrestart.js [hunk 2]
```diff
@@ -129,7 +129,7 @@ define(['kbwidget', 'bootstrap', 'jquery', 'narrativeConfig', 'common/runtime',
          * for the function.
          */
         window.KBError = function (where, what) {
-            return KBFail(false, where, what);
+            return window.KBFail(false, where, what);
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/kbaseNarrativePrestart.js:132]: keep eslint quiet

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js [hunk 3]
```diff
@@ -50,16 +51,13 @@ define([
...
 
         function makeInputControl() {
-            let selectOptions;
-            const selectElem = select(
+            return select(
                 {
                     class: 'form-control',
                     dataElement: 'input',
                 },
-                [option({ value: '' }, '')].concat(selectOptions)
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js:59]: `selectOptions` is undef, so why bother even declaring it?

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js [hunk 4]
```diff
@@ -262,9 +243,6 @@ define([
                                 callback(taxon);
                             });
                         },
-                        formatMoreResults: function () {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js:265]: not sure what this was supposed to be, but I couldn't find `formatMoreResults` in the select2 docs

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js [hunk 5]
```diff
@@ -373,7 +351,9 @@ define([
         function stop() {
             return Promise.try(() => {
                 if (container) {
-                    parent.removeChild(container);
+                    $(ui.getElement('input-container.input')).off('change');
+                    $(ui.getElement('input-container.input')).select2('destroy');
+                    container.remove();
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js:356]: clean up properly, including removing the select2 gubbins

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/select2ObjectInput.js [hunk 6]
```diff
@@ -254,21 +254,61 @@ define([
...
-                    ' (v' + objectInfo.version + ')<br>',
-                    div({ style: 'margin-left: 7px' }, [
-                        '<i>' + objectInfo.typeName + '</i><br>',
-                        'Narrative id: ' + objectInfo.wsid + '<br>',
-                        'updated ' +
-                            TimeFormat.getTimeStampStr(objectInfo.save_date) +
-                            ' by ' +
-                            objectInfo.saved_by,
-                    ]),
+                    span(
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/select2ObjectInput.js:269]: remove inline styles

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/select2ObjectInput.js [hunk 7]
```diff
@@ -431,7 +471,10 @@ define([
         function stop() {
             return Promise.try(() => {
                 if (container) {
-                    parent.removeChild(container);
+                    $(ui.getElement('input-container.input')).off('change');
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/select2ObjectInput.js:474]: clean up properly after removing the select2 element

File: test/unit/testUtil.js [hunk 8]
```diff
@@ -52,11 +52,58 @@ define('testUtil', ['bluebird', 'json!/test/testConfig.json'], (Promise, TestCon
...
+     * @returns {DOM element} the element
+     */
+    function waitForElement(documentElement, selector) {
+        return new Promise((resolve) => {
+            const element = document.querySelector(selector);
+            if (element) {
+                resolve(element);
+            }
+
+            const observer = new MutationObserver(() => {
```
Comment (ialarmedalien) [inline test/unit/testUtil.js:68]: `[MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)` tracks changes to the DOM. It keeps detailed records of what type of change has occurred, which can be queried; however, for these purposes, it is easier just to check whether the DOM element that we're interested in has appeared.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js (hunk 4 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/appWidgets2/input/taxonomyRefInput.js:265]: Might be from an older version?

**Review by briehl — APPROVED** [review]
Looks fine, seems like some setup for the proceeding PRs.

**Pull request merged**